src/services/auth/auth-lib.ts

```diff
@@ -316,3 +316,27 @@ export function getDevice(kv?: string): string {
 
 	return value;
 }
+
+
+/**
+ * Get all id of users that have a particular role within the database.
+ * @param role role that we want to filter for
+ * @returns Promise<string[]> - if valid, then contains array of user w/ role. If invalid, then contains "Unknown Error".
+ */
+export async function getUsersWithRole(role : string): Promise<string[]> {
+
+	// Makse a reference to the roles collection
+	const collection: Collection = databaseClient.db(Constants.AUTH_DB).collection(AuthDB.ROLES);
+
+	//now make a mongodb query that iterates thru and check which ones have the role
+	const queryCriteria : { roles: { $in: string[] } } = { roles: { $in: [role] } };
```

Comment:
`{ roles: { $in: string[] } }` --> Create/use a pre-defined type instead. I think you can use a `Filter<Document` instead, which makes this cleaner